src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -915,14 +919,22 @@ public async Task DisableTwoFactorProviderAsync(Organization organization, TwoFa
                         orgUser.Permissions = JsonSerializer.Serialize(invite.Permissions, JsonHelpers.CamelCase);
                     }
 
-                    if (!orgUser.AccessAll && invite.Collections.Any())
-                    {
-                        limitedCollectionOrgUsers.Add((orgUser, invite.Collections));
-                    }
-                    else
+                    var collections = invite.Collections;
+                    if (!FlexibleCollectionsIsEnabled)
                     {
-                        orgUsers.Add(orgUser);
+                        // If not using Flexible Collections - add access to all collections if user has EditAnyCollection or AccessAll permissions
+                        if (orgUser.GetPermissions()?.EditAnyCollection ?? false)
+                        {
+                            var orgCollections = await _collectionRepository.GetManyByOrganizationIdAsync(orgUser.OrganizationId);
+                            collections = orgCollections.Select(c => new CollectionAccessSelection { Id = c.Id, Manage = true });
+                        }
+                        else if (orgUser.AccessAll)
+                        {
+                            var orgCollections = await _collectionRepository.GetManyByOrganizationIdAsync(orgUser.OrganizationId);
+                            collections = orgCollections.Select(c => new CollectionAccessSelection { Id = c.Id, ReadOnly = true });
+                        }
```

Comment:
You'll probably need to revisit this per my comment above.

I think we'll need similar logic in a few places, can we extract it to like a static helper class or something? In any case - I definitely wouldn't want to nest this this far in this foreach block.

Reply:
This has been extracted into a private method